packages/apps/reputation-oracle/server/src/modules/webhook/webhook-outgoing.service.ts

```diff
@@ -138,19 +90,44 @@ export class WebhookService {
    * @param {object} payload - The data payload to send.
    * @throws {ControlledError} If the webhook request fails.
    */
-  public async sendWebhook(
-    url: string,
-    payload: Record<string, unknown>,
-  ): Promise<void> {
+  public async sendWebhook(url: string, payload: object): Promise<void> {
     const snake_case_body = CaseConverter.transformToSnakeCase(payload);
     const signedBody = await signMessage(
       snake_case_body,
       this.web3ConfigService.privateKey,
     );
-    await firstValueFrom(
+    const { status } = await firstValueFrom(
       await this.httpService.post(url, snake_case_body, {
         headers: { [HEADER_SIGNATURE_KEY]: signedBody },
       }),
     );
+
+    if (status !== HttpStatus.CREATED) {
+      throw new ControlledError(ErrorWebhook.NotSent, HttpStatus.NOT_FOUND);
+    }
+  }
+
+  public async processPendingOutgoingWebhooks(): Promise<void> {
+    const webhookEntities = await this.webhookOutgoingRepository.findByStatus(
+      WebhookOutgoingStatus.PENDING,
+    );
+
+    for (const webhookEntity of webhookEntities) {
+      try {
+        const { url, payload } = webhookEntity;
+
+        await this.sendWebhook(url, payload);
+      } catch (err) {
+        const errorId = uuidv4();
+        const failureDetail = `${ErrorWebhook.PendingProcessingFailed} (Error ID: ${errorId})`;
+        this.logger.error(
+          `Error processing pending outgoing webhook. Error ID: ${errorId}, Webhook ID: ${webhookEntity.id}, Reason: ${failureDetail}, Message: ${err.message}`,
+        );
+        await this.handleWebhookOutgoingError(webhookEntity, failureDetail);
```

Comment:
`handleWebhookOutgoingError` and similar methods in other classes now don't have to be `public` and can be tested as part of `process*` method tests